### DISCUSSION TITLE
Tips for using it with GnuRadio Soapy Custom Source

### DISCUSSION THREAD

**Discussion by ReuvenSwirsky** [cariboulabs/cariboulite#52] [General]
Just got it working in GnuRadio (3.9-Maint) with the Soapy.

- **Sample Rate must be set to 4E6 (4M). Anything else is rejected.**

Here's what you need to know for the Soapy Custom Source settings:

- Under "General"

```
- Driver: Cariboulite
- Device Arguments: device_id=1
```

- Under "RF Options"
Note that the only antenna string that works for device_id=1 is "TX/RX 6GHz" (yes, with the space in it!)

The other default options i changed 
```
Ch0: Antenna:      TX/RX 6GHz  
Ch0: Automatic DC Removal: False
Ch0 Overall Gain Value: 50
```

![20230210_233438601_iOS](https://user-images.githubusercontent.com/13056860/218222580-ced2b62a-3a7e-48e5-965a-602c5e2fef84.jpg)

![20230210_233445892_iOS](https://user-images.githubusercontent.com/13056860/218222589-2631747a-3bda-4500-a891-8d6ba0b8e3e3.jpg)

![20230210_232614277_iOS](https://user-images.githubusercontent.com/13056860/218222599-6c2c0d94-96a0-4b0b-b80d-47604a7c7a9f.jpg)

I've had it running here receiving a 2 MHz chunk of the FM Broadcast Band here, and it seems stable, with the exception that it will stop working if you change frequency.  I'm going to run it over the weekend to see if it can sustain this.

**Comment by edegraaff** [comment]
> Just got it working in GnuRadio (3.9-Maint) with the Soapy.
> 
> * **Sample Rate must be set to 4E6 (4M). Anything else is rejected.**
> 
> Here's what you need to know for the Soapy Custom Source settings:
> 
> * Under "General"
> 
> ```
> - Driver: Cariboulite
> - Device Arguments: device_id=1
> ```
> 
> * Under "RF Options"
>   Note that the only antenna string that works for device_id=1 is "TX/RX 6GHz" (yes, with the space in it!)
> 
> The other default options i changed
> 
> ```
> Ch0: Antenna:      TX/RX 6GHz  
> Ch0: Automatic DC Removal: False
> Ch0 Overall Gain Value: 50
> ```
> 
> ![20230210_233438601_iOS](https://user-images.githubusercontent.com/13056860/218222580-ced2b62a-3a7e-48e5-965a-602c5e2fef84.jpg)
> 
> ![20230210_233445892_iOS](https://user-images.githubusercontent.com/13056860/218222589-2631747a-3bda-4500-a891-8d6ba0b8e3e3.jpg)
> 
> ![20230210_232614277_iOS](https://user-images.githubusercontent.com/13056860/218222599-6c2c0d94-96a0-4b0b-b80d-47604a7c7a9f.jpg)
> 
> I've had it running here receiving a 2 MHz chunk of the FM Broadcast Band here, and it seems stable, with the exception that it will stop working if you change frequency. I'm going to run it over the weekend to see if it can sustain this.

Could you describe the used hardware, os, and prereq?

**Comment by hb9dug** [comment]
Bonjour Robert,
Have you tried the GNUradio Soapy Sink for the CaribouLite ?
On my side, GNUradio crashes when running a flow graph with the sink block... 
I run on PiSDR.

Rgds,
Michel

**Reply by konimaru** [reply]
Could this have something to do with the blocks still using **hackrf** driver IDs? Just guessing here, not being familiar with GR.
Could you send me a quick HowTo to reproduce the issue?

**Reply by PeterQFR** [reply]
I've tried the SOAPY Custom Sink, with a cariboulite running on `SoapySDRServer --bind` on the raspberry pi. 
I get the following issue that :
```
NFO] Initializing DeviceID: 0, Label: CaribouLite S1G[9170cbd6], ChannelType: S1G
[INFO] Creating SampleQueue MTU: 131072 I/Q samples (524288 bytes)
[1m[33m[WARNING] setSampleRate: using rounded rate 1333000 is deprecated; use 4e6/3 or 1333333.3.[0m
Traceback (most recent call last):
  File "/home/peter/software_defined_radio/flowgraphs/qpsk_stage1.py", line 294, in <module>
    main()
  File "/home/peter/software_defined_radio/flowgraphs/qpsk_stage1.py", line 272, in main
    tb = top_block_cls()
         ^^^^^^^^^^^^^^^
  File "/home/peter/software_defined_radio/flowgraphs/qpsk_stage1.py", line 106, in __init__
    self.soapy_custom_sink_0.set_antenna(0, 'TX/RX 6GHz')
ValueError: sink: Antenna TX/RX 6GHz at channel 0 is not supported. Available antennas are: [TX/RX Sub1GHz]
```
It seems with the above settings ie device_id being set as per @edegraaff 's good example, it defaulted to the Sub1G antenna. 

However if I change the driver in the Soapy Custom Sink to this:
`remote,device_id=1`, and leave "Device arguments" empty it will launch the HiF antenna correctly.  Using remote is how the device appears via a SoapySDRServer, these drivers are available from the SoapySDRUtil --find or --probe.
